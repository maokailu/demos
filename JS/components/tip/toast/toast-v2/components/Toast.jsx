
// 生成Notification单例并提供销毁方法
import Notification from './Notification';
let newNotification;
const getNewNotification = () => {
    // 单例 保持页面始终只有一个Notification
    if (!newNotification) {
        newNotification = Notification.reWrite();
    }
    return newNotification;
};
// 调用Notification的方法添加Notice
const addNotice = (content, duration) => {
    const notification = getNewNotification();
    notification.addNotice({
        duration,
        content
    });
};
export default {
    // 显示
    show(content, duration) {
        return addNotice(content, duration);
    },
    // 销毁
    hide() {
        if (newNotification) {
            newNotification.destroy();
            newNotification = null;
        }
    }
};
