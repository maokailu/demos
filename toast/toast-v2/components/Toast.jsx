
import React from 'react';
import classNames from 'classnames';
import Notification from './Notification';
let newNotification;

// 获得一个Notification
const getNewNotification = () => {
    // 单例
    if (!newNotification) {
        newNotification = Notification.reWrite();
    }

    return newNotification;
};

// 完成对Notification的改变
const notice = (content,duration = 100) => {
    let notificationInstance = getNewNotification();

    notificationInstance.notice({
        duration,
        content:
            <div className={
                classNames(['zby-toast-box'])
            }>
                <div className="zby-toast-content">{content}</div>
            </div>
    });
};

export default {
    // 显示
    show(content, duration) {
        return notice(content,duration);
    },
    // 销毁
    hide() {
        if (newNotification) {
            newNotification.destroy();
            newNotification = null;
        }
    }
};
