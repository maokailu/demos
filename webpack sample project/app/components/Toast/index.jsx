/**
 * Created by Aus on 2017/8/24.
 */
import Toast from './Components/Toast.jsx';

export default Toast;
