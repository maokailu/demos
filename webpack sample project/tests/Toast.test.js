import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Toast from '../app/components/Toast2';

class Test extends Component {
  constructor(props) {
    super(props);
  }
  handleToast = () =>{
    Toast.info();
  }
  render() {
    return (
      <div className="btn" onClick ={this.handleToast.bind(this)}>btn</div>
    );
  }
}
describe('A suite', () => {
  // test
  // const MockFn = jest.fn();
  jest.useFakeTimers();
  const wrapper = shallow(<Test />);
  it('Test Component should be render', function() {
    expect(wrapper.find('.btn').length).toBe(1);
  });
  it('Should render Toast with correct DOM structure', () => {
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });
  it('Simulates click events', function() {
    wrapper.find('.btn').at(0).simulate('click');
    // expect(wrapper.find('.toast').length).toBe(1);?
    // expect(MockFn).toHaveBeenCalled();
  });
  it('CloseTimer works', function() {
    jest.runAllTimers();
    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 2000);
    expect(wrapper.find('.toast').length).toBe(0);
  });
});
