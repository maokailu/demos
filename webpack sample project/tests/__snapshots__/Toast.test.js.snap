// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite Should render Toast with correct DOM structure 1`] = `null`;

exports[`A suite Should render Toast with correct DOM structure 2`] = `null`;
